lib/Sema/TypeChecker.cpp

```diff
@@ -627,6 +627,8 @@ void swift::performTypeChecking(SourceFile &SF, TopLevelContext &TLC,
   if (SF.ASTStage == SourceFile::TypeChecked)
     return;
 
+  SharedTimer timer("Type checking");
```

Comment:
This is not a change we want; we don't want to count name binding with type checking because sometimes we've already done it and sometimes we haven't.

Reply:
It's a fair point, however I do want a timer that records time spent in this function because it is not obvious to me that all the subparts are covered. The function is named "performTypeChecking", so I have renamed the timer "performTypeChecking". Is there a better name for this function and this timer?
If you like, I can change the name of both the function and the timer to performNameBindingAndTypeChecking ?

Reply:
I don't think that's correct either. I think the previous division was correct, and it's not meaningful to measure the time spent in the entire function. I also don't think "performNameBindingAndTypeChecking" would be an improvement because that *isn't* the primary function of this function. Instead, it's required that a file be name-bound before doing type-checking, and then it's been convenient for performTypeChecking to do that as needed if it hasn't already happened. At best, I would just put this in a comment; it wouldn't increase understanding at the call site to have it in the name.

("name binding" should also be renamed to "import resolution" at this point since that's all it's done for years, but that should probably be a separate PR.)

Reply:
I've removed the incorrect timer. Since, the previous division did not capture everything that was called from here, so I added a couple of new timers in callees.